### ISSUE TITLE
0.36 bate2 update error

### ISSUE THREAD

**Issue by 441088327** [CocoaPods/CocoaPods#3102]
### Command

```
/usr/local/bin/pod update
```

### Report

* What did you do?

* What did you expect to happen?

* What happened instead?

### Stack

```
   CocoaPods : 0.36.0.beta.2
        Ruby : ruby 2.2.0p0 (2014-12-25 revision 49005) [x86_64-darwin14]
    RubyGems : 2.4.5
        Host : Mac OS X 10.10.2 (14C109)
       Xcode : 6.1.1 (6A2008a)
         Git : git version 1.9.3 (Apple Git-50)
Ruby lib dir : /usr/local/Cellar/ruby/2.2.0/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ e79828498ef65642474287543feebe172348a0c4
```

### Plugins

```
cocoapods-plugins : 0.4.0
cocoapods-trunk   : 0.5.0
cocoapods-try     : 0.4.3
```

### Podfile

```ruby
xcodeproj 'DoudouApp'


target 'DoudouMusic' do
    platform :ios,7.0
    pod 'MBProgressHUD','>= 0.9'
    pod 'AFNetworking' , '>=2.5.0'
    pod 'FMDB','>=2.4'
    pod 'Reachability','>=3.2'
    pod 'DAAppsViewController','>=1.3.0'
    pod 'SSKeychain','>=1.2.2'
    #  pod 'OpenUDID','>=1.0.0'
    pod 'DOUAudioStreamer','>=0.2.9'
    pod 'LMAlertView','>=1.1.0'
    pod 'KKGestureLockView','>=1.0.0'
    #  pod 'dyci', :git => 'https://github.com/DyCI/dyci-main.git'
    pod 'SWTableViewCell', '>= 0.3.7'
    pod 'PureLayout','>=2.0.4'
    pod 'pop','>=1.0.7'
    pod 'FLEX', '>= 1.1.1'
    pod 'FLAnimatedImage', '>= 1.0.7' #GIF
    pod 'Colours', '>= 5.5.0'
    pod 'UICountingLabel', '>= 1.2.0'
    pod 'SDWebImage' ,'>=3.7.1'
    pod 'TTTAttributedLabel' , '>=1.13.0'
    
    
end


target 'DoudouNote' do
    platform :ios,7.0
    pod 'SDWebImage' ,'>=3.7.1'
    pod 'TTTAttributedLabel' , '>=1.13.0'
    pod 'MRProgress', '>= 0.8.1'
    pod 'SWTableViewCell', '>= 0.3.7'
#    pod 'SwiftyJSON', :git => 'https://github.com/SwiftyJSON/SwiftyJSON.git'
#    pod 'PathKit', :git => 'https://github.com/kylef/PathKit.git'
#    pod 'ExSwift', :git => 'https://github.com/pNre/ExSwift.git'
#    pod 'Cartography', :git => ''
end








target 'DoudouMusicToday' do
    
end
```

### Error

```
NoMethodError - undefined method `build_configurations' for nil:NilClass
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/installer.rb:707:in `configure_app_extension_api_only_for_target'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/installer.rb:471:in `block in set_target_dependencies'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/installer.rb:469:in `each'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/installer.rb:469:in `set_target_dependencies'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/installer.rb:129:in `block in generate_pods_project'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/user_interface.rb:49:in `section'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/installer.rb:125:in `generate_pods_project'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/installer.rb:93:in `install!'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/command/project.rb:71:in `run_install_with_update'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/command/project.rb:155:in `run'
/usr/local/lib/ruby/gems/2.2.0/gems/claide-0.8.0/lib/claide/command.rb:312:in `run'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/lib/cocoapods/command.rb:45:in `run'
/usr/local/lib/ruby/gems/2.2.0/gems/cocoapods-0.36.0.beta.2/bin/pod:44:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by segiddins** [comment]
@neonichu can you take a look into it?

**Comment by neonichu** [comment]
Fixed by 0505ce41c5775d22b845ea6af2f69c212230ef92

The issue was an app extension target which didn't have any pods configured. In that case, `native_target` is `nil`, but also the list of `pod_targets` is empty.
